staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -259,6 +261,14 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
+			if len(rule.FieldPath) != 0 {
+				validFieldPath, err := validFieldPath(rule.FieldPath, nil, sts)
+				if err != nil {
+					errs = append(errs, field.Invalid(fldPath, sts.Type, fmt.Sprintf("%v formatting FieldPath: %v", err, rule.FieldPath)))
+				} else {
+					fldPath = fldPath.Child(validFieldPath[1:])
```

Comment:
This doesn't produce a valid field path.  xref: https://github.com/kubernetes/kubernetes/blob/56f6030125d737dd031d77f45af6a413d1990862/staging/src/k8s.io/apimachinery/pkg/util/validation/field/path.go#L49

I think what we should do is have `validateFieldPath` take a rootFieldPath arg (which is allowed to be nil, and can be set to nil when validating CRDs), and then when validating in CRs, pass in the right rootFieldPath and then have validateFieldPath return an actual `field.Path` struct that has been constructed by calling `.Child()`, `.Key()` or `.Index()` in the parser for the correct cases.

Reply:
Updated